src/gui/modals/ExportProjectDialog.cpp

```diff
@@ -159,8 +159,8 @@ void ExportProjectDialog::startExport()
 					static_cast<AudioEngine::qualitySettings::Interpolation>(interpolationCB->currentIndex()),
 					static_cast<AudioEngine::qualitySettings::Oversampling>(oversamplingCB->currentIndex()) );
 
-	const int samplerates[5] = { 44100, 48000, 88200, 96000, 192000 };
-	const bitrate_t bitrates[6] = { 64, 128, 160, 192, 256, 320 };
+	const auto samplerates = std::array{44100, 48000, 88200, 96000, 192000};
+	const auto bitrates = std::array{64, 128, 160, 192, 256, 320};
```

Comment:
I disagree with this. Leaving the template arguments to be automatically deduced is similar to using `auto` to declare variables, which we already try to do and has numerous benefits. The Core Guidelines [recommend automatically deducing template arguments](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#t44-use-function-templates-to-deduce-class-template-argument-types-where-feasible) (the guideline is a bit out of date, as it's written for C++14, but the concept is the same).

I don't think anyone would have complained if `bitrates` were an `int[]` originally (indeed, `samplerates` could use `sample_rate_t`). If it's that important to distinguish the types, then some sort of strict typedef (such as `enum class bitrate_t : std::uint16_t { };`) should be used, at which point specifying the element type alone won't work anyway.

In fact, there are a number of places in this PR where what used to be a `float[]` (or even a `double[]`) is now a `std::array<sample_t, N>` thanks to automatically deduced types, which is more appropriate for the situation and can avoid a runtime conversion.

Reply:
I honestly never really considered the pros and cons of either approach. Dom brought to my attention that I can simplify the construction of these types using template deduction initially, which made the code simpler. There are other places in the code where I mixed the two approaches together, so I'm planning on just deducing wherever possible. I also try to follow the guidelines whenever possible. Going against it is probably not a good idea in the long run. It also helps to deduce in this case because the size doesn't have to be specified every time  

I still don't see this approach as "bad", per-se. The guidelines said that "Writing the template argument types explicitly can be tedious and unnecessarily verbose". I don't think specifying the type and the size made the code more verbose to the point where deducing is a hard recommendation.